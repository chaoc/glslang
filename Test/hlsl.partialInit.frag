struct outs {
    int a;
    float b;
    bool c;
    float4 v;
};

static float4 gv = {0,0,1};
static float gfa[3] = {0,0};

outs PixelShaderFunction(float4 input) : COLOR0
{
    outs o2 = { 3 };
    outs o4;
    o4.v = gv * gfa[2];
    outs o1 = { };
//    outs o3 = (outs)0;
//    o4 = (outs)0;
    o4.c = o1.c;

    return o4;
}