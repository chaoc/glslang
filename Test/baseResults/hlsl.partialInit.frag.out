hlsl.partialInit.frag
Shader version: 450
gl_FragCoord origin is upper left
0:? Sequence
0:8  Sequence
0:8    move second child to first child (temp 4-component vector of float)
0:8      'gv' (global 4-component vector of float)
0:8      Constant:
0:8        0.000000
0:8        0.000000
0:8        1.000000
0:8        0.000000
0:9  Sequence
0:9    move second child to first child (temp 3-element array of float)
0:9      'gfa' (global 3-element array of float)
0:9      Constant:
0:9        0.000000
0:9        0.000000
0:9        0.000000
0:12  Function Definition: PixelShaderFunction(vf4; (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:12    Function Parameters: 
0:12      'input' (layout(location=0 ) in 4-component vector of float)
0:?     Sequence
0:13      Sequence
0:13        move second child to first child (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:13          'o2' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:13          Constant:
0:13            3 (const int)
0:13            0.000000
0:13            false (const bool)
0:13            0.000000
0:13            0.000000
0:13            0.000000
0:13            0.000000
0:15      move second child to first child (temp 4-component vector of float)
0:15        v: direct index for structure (temp 4-component vector of float)
0:15          'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:15          Constant:
0:15            3 (const int)
0:15        vector-scale (temp 4-component vector of float)
0:15          'gv' (global 4-component vector of float)
0:15          direct index (temp float)
0:15            'gfa' (global 3-element array of float)
0:15            Constant:
0:15              2 (const int)
0:16      Sequence
0:16        move second child to first child (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:16          'o1' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:16          Constant:
0:16            0 (const int)
0:16            0.000000
0:16            false (const bool)
0:16            0.000000
0:16            0.000000
0:16            0.000000
0:16            0.000000
0:19      move second child to first child (temp bool)
0:19        c: direct index for structure (temp bool)
0:19          'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:19          Constant:
0:19            2 (const int)
0:19        c: direct index for structure (temp bool)
0:19          'o1' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:19          Constant:
0:19            2 (const int)
0:21      Sequence
0:21        Sequence
0:21          move second child to first child (temp int)
0:?             'a' (layout(location=0 ) out int)
0:21            a: direct index for structure (temp int)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                0 (const int)
0:21          move second child to first child (temp float)
0:?             'b' (layout(location=1 ) out float)
0:21            b: direct index for structure (temp float)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                1 (const int)
0:21          move second child to first child (temp bool)
0:?             'c' (layout(location=2 ) out bool)
0:21            c: direct index for structure (temp bool)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                2 (const int)
0:21          move second child to first child (temp 4-component vector of float)
0:?             'v' (layout(location=3 ) out 4-component vector of float)
0:21            v: direct index for structure (temp 4-component vector of float)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                3 (const int)
0:21        Branch: Return
0:?   Linker Objects
0:?     'a' (layout(location=0 ) out int)
0:?     'b' (layout(location=1 ) out float)
0:?     'c' (layout(location=2 ) out bool)
0:?     'v' (layout(location=3 ) out 4-component vector of float)
0:?     'input' (layout(location=0 ) in 4-component vector of float)
0:?     'gv' (global 4-component vector of float)
0:?     'gfa' (global 3-element array of float)


Linked fragment stage:


Shader version: 450
gl_FragCoord origin is upper left
0:? Sequence
0:8  Sequence
0:8    move second child to first child (temp 4-component vector of float)
0:8      'gv' (global 4-component vector of float)
0:8      Constant:
0:8        0.000000
0:8        0.000000
0:8        1.000000
0:8        0.000000
0:9  Sequence
0:9    move second child to first child (temp 3-element array of float)
0:9      'gfa' (global 3-element array of float)
0:9      Constant:
0:9        0.000000
0:9        0.000000
0:9        0.000000
0:12  Function Definition: PixelShaderFunction(vf4; (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:12    Function Parameters: 
0:12      'input' (layout(location=0 ) in 4-component vector of float)
0:?     Sequence
0:13      Sequence
0:13        move second child to first child (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:13          'o2' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:13          Constant:
0:13            3 (const int)
0:13            0.000000
0:13            false (const bool)
0:13            0.000000
0:13            0.000000
0:13            0.000000
0:13            0.000000
0:15      move second child to first child (temp 4-component vector of float)
0:15        v: direct index for structure (temp 4-component vector of float)
0:15          'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:15          Constant:
0:15            3 (const int)
0:15        vector-scale (temp 4-component vector of float)
0:15          'gv' (global 4-component vector of float)
0:15          direct index (temp float)
0:15            'gfa' (global 3-element array of float)
0:15            Constant:
0:15              2 (const int)
0:16      Sequence
0:16        move second child to first child (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:16          'o1' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:16          Constant:
0:16            0 (const int)
0:16            0.000000
0:16            false (const bool)
0:16            0.000000
0:16            0.000000
0:16            0.000000
0:16            0.000000
0:19      move second child to first child (temp bool)
0:19        c: direct index for structure (temp bool)
0:19          'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:19          Constant:
0:19            2 (const int)
0:19        c: direct index for structure (temp bool)
0:19          'o1' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:19          Constant:
0:19            2 (const int)
0:21      Sequence
0:21        Sequence
0:21          move second child to first child (temp int)
0:?             'a' (layout(location=0 ) out int)
0:21            a: direct index for structure (temp int)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                0 (const int)
0:21          move second child to first child (temp float)
0:?             'b' (layout(location=1 ) out float)
0:21            b: direct index for structure (temp float)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                1 (const int)
0:21          move second child to first child (temp bool)
0:?             'c' (layout(location=2 ) out bool)
0:21            c: direct index for structure (temp bool)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                2 (const int)
0:21          move second child to first child (temp 4-component vector of float)
0:?             'v' (layout(location=3 ) out 4-component vector of float)
0:21            v: direct index for structure (temp 4-component vector of float)
0:21              'o4' (temp structure{temp int a, temp float b, temp bool c, temp 4-component vector of float v})
0:21              Constant:
0:21                3 (const int)
0:21        Branch: Return
0:?   Linker Objects
0:?     'a' (layout(location=0 ) out int)
0:?     'b' (layout(location=1 ) out float)
0:?     'c' (layout(location=2 ) out bool)
0:?     'v' (layout(location=3 ) out 4-component vector of float)
0:?     'input' (layout(location=0 ) in 4-component vector of float)
0:?     'gv' (global 4-component vector of float)
0:?     'gfa' (global 3-element array of float)

// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 66

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "PixelShaderFunction" 45 50 56 60 65
                              ExecutionMode 4 OriginUpperLeft
                              Name 4  "PixelShaderFunction"
                              Name 9  "gv"
                              Name 17  "gfa"
                              Name 21  "outs"
                              MemberName 21(outs) 0  "a"
                              MemberName 21(outs) 1  "b"
                              MemberName 21(outs) 2  "c"
                              MemberName 21(outs) 3  "v"
                              Name 23  "o2"
                              Name 28  "o4"
                              Name 37  "o1"
                              Name 45  "a"
                              Name 50  "b"
                              Name 56  "c"
                              Name 60  "v"
                              Name 65  "input"
                              Decorate 45(a) Location 0
                              Decorate 50(b) Location 1
                              Decorate 56(c) Location 2
                              Decorate 60(v) Location 3
                              Decorate 65(input) Location 0
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 4
               8:             TypePointer Private 7(fvec4)
           9(gv):      8(ptr) Variable Private
              10:    6(float) Constant 0
              11:    6(float) Constant 1065353216
              12:    7(fvec4) ConstantComposite 10 10 11 10
              13:             TypeInt 32 0
              14:     13(int) Constant 3
              15:             TypeArray 6(float) 14
              16:             TypePointer Private 15
         17(gfa):     16(ptr) Variable Private
              18:          15 ConstantComposite 10 10 10
              19:             TypeInt 32 1
              20:             TypeBool
        21(outs):             TypeStruct 19(int) 6(float) 20(bool) 7(fvec4)
              22:             TypePointer Function 21(outs)
              24:     19(int) Constant 3
              25:    20(bool) ConstantFalse
              26:    7(fvec4) ConstantComposite 10 10 10 10
              27:    21(outs) ConstantComposite 24 10 25 26
              30:     19(int) Constant 2
              31:             TypePointer Private 6(float)
              35:             TypePointer Function 7(fvec4)
              38:     19(int) Constant 0
              39:    21(outs) ConstantComposite 38 10 25 26
              40:             TypePointer Function 20(bool)
              44:             TypePointer Output 19(int)
           45(a):     44(ptr) Variable Output
              46:             TypePointer Function 19(int)
              49:             TypePointer Output 6(float)
           50(b):     49(ptr) Variable Output
              51:     19(int) Constant 1
              52:             TypePointer Function 6(float)
              55:             TypePointer Output 20(bool)
           56(c):     55(ptr) Variable Output
              59:             TypePointer Output 7(fvec4)
           60(v):     59(ptr) Variable Output
              64:             TypePointer Input 7(fvec4)
       65(input):     64(ptr) Variable Input
4(PixelShaderFunction):           2 Function None 3
               5:             Label
          23(o2):     22(ptr) Variable Function
          28(o4):     22(ptr) Variable Function
          37(o1):     22(ptr) Variable Function
                              Store 9(gv) 12
                              Store 17(gfa) 18
                              Store 23(o2) 27
              29:    7(fvec4) Load 9(gv)
              32:     31(ptr) AccessChain 17(gfa) 30
              33:    6(float) Load 32
              34:    7(fvec4) VectorTimesScalar 29 33
              36:     35(ptr) AccessChain 28(o4) 24
                              Store 36 34
                              Store 37(o1) 39
              41:     40(ptr) AccessChain 37(o1) 30
              42:    20(bool) Load 41
              43:     40(ptr) AccessChain 28(o4) 30
                              Store 43 42
              47:     46(ptr) AccessChain 28(o4) 38
              48:     19(int) Load 47
                              Store 45(a) 48
              53:     52(ptr) AccessChain 28(o4) 51
              54:    6(float) Load 53
                              Store 50(b) 54
              57:     40(ptr) AccessChain 28(o4) 30
              58:    20(bool) Load 57
                              Store 56(c) 58
              61:     35(ptr) AccessChain 28(o4) 24
              62:    7(fvec4) Load 61
                              Store 60(v) 62
                              Return
                              FunctionEnd
